plugin/pkg/admission/limitranger/admission.go

```diff
@@ -612,3 +549,142 @@ func PodValidateLimitFunc(limitRange *corev1.LimitRange, pod *api.Pod) error {
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+type podResourcesOptions struct {
+	// InPlacePodVerticalScalingEnabled indicates that the in-place pod vertical scaling feature gate is enabled.
+	InPlacePodVerticalScalingEnabled bool
+}
+
+// podRequests is a simplified version of pkg/api/v1/resource/PodRequests that operates against the core version of
```

Comment:
Since that is fragile, it might be desirable to export a pkg/api/v1/resource function that returns a set of test scenarios that are executed in both the source package unit test and in this package's unit test.  

Did we consider instead doing a partial conversion of the pod to a v1 type (copying only key attributes) and then using the existing helper?  Several other admission plugins do something similar IIRC (quota did at one point), and if we limited the scope of what was copied we would minimize allocations in this path (probably the list of init/regular containers, their statuses are sufficient, but we'd want to be careful).

Reply:
I’m ok with this tracked as a follow up

Reply:
added TODO comments